build/avalanchego-apis/contract-chain-c-chain-api.md

```diff
@@ -742,3 +742,190 @@ curl -X POST --data '{
 }
 ```
 
+# Admin API
+
+This API can be used for debugging. Note that the Admin API is disabled by default. To run a node with the Admin API enabled, use [command line argument](../references/command-line-interface.md#c-chain-config) `--coreth-admin-api-enabled:true`.
+
+## Format
+
+This API uses the `json 2.0` RPC format.
+
+{% page-ref page="issuing-api-calls.md" %}
+
+## Endpoint
+
+```text
+/ext/bc/C/admin
+```
+
+## API Methods
+
+### admin.setLogLevel
+
+Sets the log level of the C-Chain.
+
+#### **Signature**
+
+```text
+admin.setLogLevel({level:string}) -> {success:bool}
+```
+
+* `level` is the log level to be set.
+
+#### **Example Call**
+
+```bash
+curl -X POST --data '{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "method" :"admin.setLogLevel",
+    "params": {
+        "level":"info",
+    }
+}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/C/admin
+```
+
+#### **Example Response**
+
+```javascript
+{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "result" :{
+        "success":true
+    }
+}
+```
+
+### admin.startCPUProfiler
+
+Starts a CPU profile writing to a file in the `coreth_performance_c` directory
+
+#### **Signature**
+
+```text
+admin.startCPUProfiler() -> {success:bool}
+```
+
+#### **Example Call**
+
+```bash
+curl -X POST --data '{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "method" :"admin.startCPUProfiler",
+    "params": {}
+}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/C/admin
+```
+
+#### **Example Response**
+
+```javascript
+{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "result" :{
+        "success":true
+    }
+}
+```
+
+### admin.stopCPUProfiler
+
+Stops the CPU profile
+
+#### **Signature**
+
+```text
+admin.stopCPUProfiler() -> {success:bool}
+```
+
+#### **Example Call**
+
+```bash
+curl -X POST --data '{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "method" :"admin.stopCPUProfiler",
+    "params": {}
+}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/C/admin
+```
+
+#### **Example Response**
+
+```javascript
+{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "result" :{
+        "success":true
+    }
+}
+```
+
+### admin.memoryProfile
+
+Runs a memory profile writing to a file in the `coreth_performance_c` directory.
+
+#### **Signature**
+
+```text
+admin.memoryProfile() -> {success:bool}
+```
+
+#### **Example Call**
+
+```bash
+curl -X POST --data '{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "method" :"admin.setLogLevel",
+    "params": {}
+}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/C/admin
+```
+
+#### **Example Response**
+
+```javascript
+{
+    "jsonrpc":"2.0",
+    "id"     :1,
+    "result" :{
+        "success":true
+    }
+}
+```
+
+### admin.lockProfile
+
+Runs a mutex profile writing to the `coreth_performance_c` directory.
```

Comment:
@danlaine so it writes to this directory https://github.com/ava-labs/coreth-internal/blob/master/plugin/evm/vm.go#L691. I'm not sure how to word it perfectly